agent/help_flag_test.go

```diff
@@ -24,6 +24,18 @@ var _ = Describe("Help flag for host agent", func() {
 				"-namespace string",
 				"-skip-installation",
 				"-version",
+				"-add_dir_header",
```

Comment:
I might be missing context, but do we need a whole bunch of these?

Reply:
Enabling the `klog` based verbosity automatically adds these flags https://github.com/kubernetes/klog/blob/521139aabdab1ebae1871e82e48928b1b4a8a562/klog.go#L402, 

Checked with `kubeadm` they are also supporting the above flags.

Reply:
To activate the `"v"` flag we need to add `klog` and `klog.Initflags(nil)`, which in turn adds the other flags too. I think #311 removes the extra flags by removing exactly these two code blocks from `main.go`. I am not sure how to enable only the `"v"` flag without enabling the other ones.

Reply:
[kubernetes/kubeadm](https://github.com/kubernetes/kubernetes/blob/4033355d292505eaed05655bd91f4ea9edc7ef27/cmd/kubeadm/app/kubeadm.go#L32) uses `pflag` to hide these command-line arguments. Should we try something similar?

Reply:
This is a good find. We could use this to explicitly hide some of the flags.

Reply:
However, a possible downside is this adds a new dependency: github.com/spf13/pflag

Reply:
In my mind, it should be okay as it is already there as a transitive dependency in [go.sum](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/go.sum#L1092)